internal/controllers/v1beta1/gateway/gateway.go

```diff
@@ -168,6 +176,11 @@ func (c *GatewayController) CreateCertificate(ctx context.Context, gateway *netw
 			},
 		},
 	}
+
+	if tcb {
```

Comment:
alternative nit: The conditional check on L149 could be entirely moved here and remove the need for the `tcb` variable.